cashu/src/main/java/org/signal/cashu/util/CryptoUtil.kt

```
@@ -0,0 +1,90 @@
+            val signature = proof.id.hexToBytes()
+
+            // Verify signature
+            val signer = ECDSASigner()
+            val domainParams = org.bouncycastle.crypto.params.ECDomainParameters(curve.curve, curve.g, curve.n, curve.h)
+            signer.init(false, ECPublicKeyParameters(curve.g, domainParams))
+
+            val r = BigInteger(1, signature.copyOfRange(0, 32))
+            val s = BigInteger(1, signature.copyOfRange(32, 64))
+
+            return signer.verifySignature(commitment.getEncoded(true), r, s)
+        } catch (e: Exception) {
```

detekt (1.23.7)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)